Tab-complete relative partest paths (eg pos/t<TAB>)

dwijnand opened the pull request:
For example:

    > partest pos/t17<TAB>
    pos/t1711                 pos/t1722-A.scala         pos/t1748.scala           pos/t1751                 pos/t177.scala            pos/t1785.scala           pos/t1786-cycle.scala     pos/t1798.scala
    pos/t1722                 pos/t1745                 pos/t175.scala            pos/t1756.scala           pos/t1782                 pos/t1786-counter.scala   pos/t1789.scala

I wrote a little build test, which you can manually run (see commit for instructions).

Fixes https://github.com/scala/bug/issues/11184

SethTisue commented:
if I `partest run<TAB>` I get `partest runtest/files/run/`

also it doesn't seem to work for other partest "roots" besides `test/files`, so for example I don't seem to be able to tab-complete my way into `test/tasty/neg`

(but the PR is progress and seems mergeable regardless)

dwijnand commented:
Hmm.  Also `partest pos/t177<TAB>` -> `partest pos/t177.scala` (good)
and then `partest pos/t177.scala<TAB>` -> `partest pos/t177.scalatest/files/pos/t177.scala` (terribad)

I'll look.

dwijnand committed:
Tab-complete relative partest paths (eg pos/t<TAB>)

For example:

    > partest pos/t17<TAB>
    pos/t1711                 pos/t1722-A.scala         pos/t1748.scala           pos/t1751                 pos/t177.scala            pos/t1785.scala           pos/t1786-cycle.scala     pos/t1798.scala
    pos/t1722                 pos/t1745                 pos/t175.scala            pos/t1756.scala           pos/t1782                 pos/t1786-counter.scala   pos/t1789.scala

I wrote a little build test, which you can manually run.

dwijnand commented:
> also it doesn't seem to work for other partest "roots" besides `test/files`, so for example I don't seem to be able to tab-complete my way into `test/tasty/neg`

Just noticed this.  This is also the unfortunate behaviour of status quo (i.e. not a regression I'm introducing).  I'm happy to push fixing that as I've sent too long on this already.

SethTisue commented:
the new test doesn't get run, so before merge, I think this needs comments that:
* emphasize it doesn't run automatically
* say how to run it manually

SethTisue commented:
(maybe in a readme in the new directory?)
